tap_zuora/client.py

```diff
@@ -51,19 +81,25 @@ def __init__(
         self,
         username: str,
         password: str,
+        auth_type: str,
         partner_id: str,
         sandbox: bool = False,
         european: bool = False,
         is_rest: bool = False,
     ):
         self.username = username
         self.password = password
+        self.auth_type = auth_type
         self.sandbox = sandbox
         self.european = european
         self.partner_id = partner_id
         self.is_rest = is_rest
         self._session = requests.Session()
 
+        if self.auth_type == "Basic":
+            self.access_token = None
+        else:
+            self.access_token = get_access_token(self.username, self.password, URLS[(self.sandbox, self.european)])
```

Comment:
```suggestion
        if self.auth_type == "Basic":
            self.access_token = None
        elif self.auth_type == "OAuth":
            self.access_token = get_access_token(self.username, self.password, URLS[(self.sandbox, self.european)])
        else:
            raise Exception("...")
```

Reply:
I think we should have auth_type validation in the constructor as well to make it more fail safe in case we don't invoke it from [here](https://github.com/singer-io/tap-zuora/pull/79/files#diff-dd2afebb2acbc536da53cd1eb37ca3716679ab935b9889aac034441ac16b402aR115).

Reply:
Got it. Add validation in constructor as well